api/users.rb

```diff
@@ -77,3 +77,14 @@
   user.mark_as_read(source)
   user.reload.to_hash.to_json
 end
+
+post "#{APIPREFIX}/users/:user_id/retire" do |user_id|
+  return {}.to_json if not params["retired_username"]
+  user = User.find_by!(external_id: user_id)
+  user.update_attribute(:email, "")
+  user.update_attribute(:notification_ids, [])
+  user.update_attribute(:read_states, [])
+  user.unsubscribe_all
+  user.retire_all_comments(params["retired_username"])
+  user.update_attribute(:username, params["retired_username"])
```

Comment:
I think we should also set the user model's field username to the retired username as well so that exports that may happen at the database level are fully scrubbed.

Reply:
@macdiesel Correct - that's what this line will accomplish. Is there a different document field that I've missed?
I had to use a bit of a backdoor solution to change "author_username" in the contents collection - I'll point that out below.

Reply:
I think I misinterpreted this. Carry on.